pkg/deployments/scaler.go

```diff
@@ -14,13 +14,37 @@ type scaler struct {
 	minScale     int32
 	maxScale     int32
 
+	lastScaleDown time.Time
+
 	// scaleFuncMtx ensures the scale function is not run concurrently.
 	scaleFuncMtx sync.Mutex
 	scaleFunc    func(n int32, atLeastOne bool) error
 
-	scaleDownDelay   time.Duration
-	scaleDownStarted bool
-	scaleDownTimer   *time.Timer
+	scaleDownDelay time.Duration
+}
+
+func newScaler(scaleDownDelay time.Duration, scaleFunc func(int32, bool) error) *scaler {
+	s := &scaler{
+		// -1 represents unknown
+		currentScale:   -1,
+		desiredScale:   -1,
+		scaleDownDelay: scaleDownDelay,
+		lastScaleDown:  time.Now(),
```

Comment:
Use `time.Time{}` (empty time) for this.

Reply:
I think just not initializing does the same thing? Can you verify the latest code? it seems to all work as expected right now.